src/sweeper/sweepers/duplicates.py

```diff
@@ -1,65 +1,91 @@
 #!/usr/bin/env python
 # * coding: utf8 *
-'''
-duplicates.py
-A module that removes duplicate geometry or attributes or both
-'''
-
-import arcpy
 import re
-import os
+
 from xxhash import xxh64
 
-class DuplicateTest(object):
+import arcpy
+
+
+class DuplicateTest():
+    '''A class that finds and removes duplicate geometries or attributes or both
+    '''
     def __init__(self, workspace, table_name):
         self.report = {}
         self.workspace = workspace
         self.table_name = table_name
 
 
     def sweep(self):
+        '''A method that finds duplicate records and returns a report dictionary
+        '''
+        digests = set([])
+
+        truncate_shape_precision = re.compile(r'(\d+\.\d{2})(\d+)')
+
+        with arcpy.EnvManager(workspace=self.workspace):
+            description = arcpy.da.Describe(self.table_name)
```

Comment:
```suggestion
            description = arcpy.Describe(self.table_name)
```

Reply:
Isn’t da the new describe namespace?

Reply:
Did this fail to run in python 3 @gregbunce?

Reply:
Yup, it did fail in python3.  ?  I removed the da, and all was good.

Reply:
Hmm ok. I though da was the new hotness.

Reply:
I'm pretty sure that `da` is still there in Pro and is probably more performant: https://pro.arcgis.com/en/pro-app/arcpy/data-access/what-is-the-data-access-module-.htm

@gregbunce What error message did you see when you tried to run it in python 3?

Reply:
@stdavis it was returning a dictionary error - description.hasOID would fail but description['hasOID'] would return True

Reply:
Yes, that would be a difference between the old `Describe` and the newer `da.Describe`. You have to access the data from a dictionary rather than an object.

Reply:
Let’s switch back to the da namespace and the dictionary access style. I could use a little help so feel free to update that or merge and update another branch

Reply:
out of curiosity... what is the benefit of using arcpy.Describe over using the field type property in arcpy.ListFields? It seems like 

fields = [field.name for field in arcpy.ListFields(self.table_name) if field.type not in ['OID', 'Guid', 'GlobalID', 'Geometry']]

accomplishes the same thing without a bunch of if statements and seems cleaner (at least to a geographer).

Reply:
Mainly it handles all the cases where the global Id and objectid and geometry aren’t using standard field names. If you can infer the names from the describe it is safer and smarter.

Reply:
I think that @ZachBeck's method doesn't depend on field names at all. It checks for field types. But I could be missing something...

Reply:
Well I missed that completely.

Reply:
yeah, I'm using the field types.

Reply:
I think for oid, global, and geometry that works well. 

I have an issue with guid as I commented above https://github.com/agrc/sweeper/pull/18#discussion_r330708709. You wouldn't have a duplicate record if a records related data was unique and the keys were built using guids. Ignoring all guids you could flag records like this.